luaui/RmlWidgets/rml_starter/rml_starter.lua

```diff
@@ -0,0 +1,184 @@
+        author = "Mupersega",
+        date = "2025",
+        license = "GNU GPL, v2 or later",
+        layer = -10000,
+        enabled = false,
+    }
+end
+
+-- Constants
+local WIDGET_ID = "rml_starter"
+local MODEL_NAME = "rml_starter_model"
+local RML_PATH = "luaui/rmlwidgets/rml_starter/rml_starter.rml"
+
+-- Widget state
+local document
+local dm_handle
+
+-- Create a new data model every time to avoid reference oddities even with dm_handle
+local function initModel()
+    return {
+
+        -- Main widget states
+        expanded = true,
+        debugMode = false,
+        activeTab = "landing",
+
+        -- All tabs
+        tabs = {
+            { id = "landing", label = "Welcome" },
+            { id = "getting-started", label = "Getting Started" },
+            { id = "base-widget-conventions", label = "Base Widget Conventions" },
+            { id = "widget-positioning", label = "Widget Positioning" },
+            { id = "data-binding", label = "Data Binding" },
+            { id = "tools", label = "Tools" },
+        },
+        
+        -- Custom class groups for this widget for repeatability
+        my = {
+            codeBlock = "flex flex-col p-3 bg-darker rounded border border-dark-alpha code-green text-sm",
+            tabsNavigationStyles = "font-bold bg-darkest-semi-alpha bg-gradient-darker-alpha radial-focus-start text-outline-darkest-lg border-bottom border-darkest",
+        },
+
+        -- Data binding demo variables
+        playerName = "Commander",
+        currentTime = os.date("%H:%M:%S"),
+
+        testArray = {
+            { name = "Configuration", value = 100 },
+            { name = "Game State", value = 200 },
+            { name = "UI Controls", value = 300 },
+            { name = "User Preferences", value = 400 },
+        },
+
+        -- Data binding examples table
+        dataBindingExamples = {
+            
+            -- Array examples for iteration
+            playerList = {
+                { name = "Player1", team = "Armada", score = 1250 },
+                { name = "Player2", team = "Cortex", score = 980 },
+                { name = "Player3", team = "Legion", score = 1100 },
+            },
+            
+            unitQueue = {
+                { name = "Construction Bot", cost = 100, time = "15s" },
+                { name = "Light Laser Turret", cost = 250, time = "30s" },
+                { name = "Solar Collector", cost = 150, time = "20s" },
+            },
+
+            availableThemes = {
+                { id = "base", name = "Base" },
+                { id = "armada", name = "Armada" },
+                { id = "cortex", name = "Cortex" },
+                { id = "legion", name = "Legion" },
+            },
+        },
+
+        -- How to cleanly use functions in the data model, use utils.GetCurrentModel(dm_handle) to avoid callback reference issues
+        setActiveTab = function(event, tabId)
+            local model = utils.GetCurrentModel(dm_handle)
```

Comment:
I feel like this a enormous pitfall for new developers as this is really unintuitive because developers expect to be able to use the dm_handle. Maybe we need to look at the engine implemenation if we can find the issue that will let us avoid having to use this workaround.